tests/device_tests/common.py

```diff
@@ -76,6 +96,61 @@ def setup_mnemonic_pin_passphrase(self, lock=True):
         self._setup_mnemonic(pin=TrezorTest.pin4, passphrase=True, lock=lock)
 
 
+SCREENSHOT_PATH = (Path(__file__) / "../../../core/src").resolve()
+SCREENSHOT_FIXTURE_PATH = (Path(__file__) / "../../ui_tests").resolve()
+
+
+def get_test_id(method):
+    return "{}_{}".format(method.__self__.__class__.__name__, method.__name__)
+
+
+def collect_images():
+    return list(SCREENSHOT_PATH.glob("*.png"))
+
+
+def collect_fixtures(test_id):
+    return list(SCREENSHOT_FIXTURE_PATH.glob("{}/*.png".format(test_id)))
+
+
+def assert_images(images, test_id):
+    fixtures = collect_fixtures(test_id)
+    if not fixtures:
+        return
+    images = sorted(images)
+    fixtures = sorted(fixtures)
+
+    for fixture, image in itertools.zip_longest(fixtures, images):
+        assert fixture is not None, "Missing fixture for image {}".format(image)
+        assert image is not None, "Missing image for fixture {}".format(fixture)
+        assert filecmp.cmp(fixture, image), "Image {} and fixture {} differ".format(
+            image, fixture
+        )
+
+
+def record_fixtures(images, test_id):
+    fixture_dir = SCREENSHOT_FIXTURE_PATH / test_id
+
+    if fixture_dir.is_dir():
+        # remove old fixtures
+        remove_files(collect_fixtures(test_id))
+    else:
+        # create the fixture dir, if not present
+        print("Creating", fixture_dir)
+        fixture_dir.mkdir()
+
+    # move the recorded images into the fixture locations
+    for index, image in enumerate(images):
```

Comment:
images are not sorted
i think this is what was causing the problem, not the enumeration